crates/rslint_parser/src/syntax/expr.rs

```diff
@@ -977,15 +981,18 @@ pub fn primary_expr(p: &mut Parser) -> Option<CompletedMarker> {
 					// async (foo, bar, ...baz) => foo
 					let m = p.start();
 					p.bump_remap(T![async]);
-					if p.at(T!['(']) {
-						parameter_list(p);
-					} else {
+					let parsed_parameters = parse_parameter_list(p);
+					if parsed_parameters.is_absent() {
 						let m = p.start();
 						// test_err async_arrow_expr_await_parameter
 						// let a = async await => {}
 						p.bump_remap(T![ident]);
 						m.complete(p, JS_IDENTIFIER_BINDING);
+					} else {
+						parsed_parameters
+							.or_missing_with_error(p, js_parse_error::expected_parameters);
```

Comment:
You won't need the `or_missing_with_error` because the `if parsed_parameters.is_absent` already covers the case where it is missing.
```suggestion
					let parsed_parameters = parse_parameter_list(p);
					if parsed_parameters.is_absent() {
						let m = p.start();
						// test_err async_arrow_expr_await_parameter
						// let a = async await => {}
						p.bump_remap(T![ident]);
						m.complete(p, JS_IDENTIFIER_BINDING);
					} 
```